### PR TITLE
Renaming records to docs

### PR DESCRIPTION
**Pull request by evanxg852000**
### Description
Closes https://github.com/quickwit-inc/quickwit/issues/534

### How was this PR tested?
unit test

### CODE REVIEW

File: quickwit-metastore/src/metastore/mod.rs [hunk 1]
```diff
@@ -68,9 +68,9 @@ pub struct SplitMetadata {
     /// in the storage URI resolver: for instance, the Amazon S3 region.
     pub split_id: String,
 
-    /// Number of records (or documents) in the split.
+    /// Number of documents in the split.
     /// TODO make u64
-    pub num_records: usize,
+    pub num_docs: usize,
```
Comment (fulmicoton) [inline quickwit-metastore/src/metastore/mod.rs:73]: That won't be backward compatible

**Comment by fulmicoton** [comment]
This change requires to PR #804

File: quickwit-metastore/src/metastore/mod.rs (hunk 1 above)
Comment (evanxg852000) [reply quickwit-metastore/src/metastore/mod.rs:73]: Yes, and this is why I was mentioning migration script option [here](https://github.com/quickwit-inc/quickwit/issues/799#issuecomment-972601268). You have already been thinking of this so I will get those changes here to see.

Comment (guilload) [reply quickwit-metastore/src/metastore/mod.rs:73]: `#[serde(alias = "num_records")]` to the rescue.

**Review by fulmicoton — APPROVED** [review]

**Pull request merged**